src/CalcManager/CalculatorHistory.cpp

```diff
@@ -35,16 +35,13 @@ unsigned int CalculatorHistory::AddToHistory(_In_ shared_ptr<CalculatorVector <p
 
 unsigned int CalculatorHistory::AddItem(_In_ shared_ptr<HISTORYITEM> const &spHistoryItem)
 {
-    int lastIndex;
-
     if (m_historyItems.size() >= m_maxHistorySize)
     {
         m_historyItems.erase(m_historyItems.begin());
     }
 
     m_historyItems.push_back(spHistoryItem);
-    lastIndex = static_cast<unsigned>(m_historyItems.size() - 1);
-    return lastIndex;
+    return static_cast<unsigned>(m_historyItems.size() - 1);
```

Comment:
On this one, I don't fully agree with the change.  Having the named variable there gives "documentation through code" that the _intention_ is to be returning the last index.  The change that _would_ make sense here would be to combine the variable declaration and assignment together:

```
int lastIndex = static_cast<unsigned>(m_historyItems.size() - 1);
return lastIndex;
```

Reply:
@HowardWolosky you might want to check a feature of GitHub that would make your suggestion applicable with one click via webui: https://help.github.com/en/articles/incorporating-feedback-in-your-pull-request#applying-a-suggested-change

Reply:
Thanks for the suggestion, @janisozaur  -- It looks like @Maharramoff hasn't configured this PR to accept edits from maintainers, unless I'm misunderstanding how the feature works.

Reply:
> Thanks for the suggestion, @janisozaur -- It looks like @Maharramoff hasn't configured this PR to accept edits from maintainers, unless I'm misunderstanding how the feature works.

@HowardWolosky  I already did it. 
![G5_l7zM8Te_VFwGRV_KzAg](https://user-images.githubusercontent.com/9804406/54560578-15453280-49dc-11e9-9d34-5ffeaa499c96.png)

Reply:
@HowardWolosky sorry, I linked to the receiving part of GitHub's review suggestions, I should've rather linked to this instead: 
[![suggestion-block](https://user-images.githubusercontent.com/550290/54562802-fd63b380-49c7-11e9-9dc5-931a8e63c0f1.png)](https://help.github.com/en/articles/commenting-on-a-pull-request) and you can see a sample of how it would like below:
```suggestion
    int lastIndex = static_cast<unsigned>(m_historyItems.size() - 1);
    return lastIndex;
```

Reply:
> On this one, I don't fully agree with the change. Having the named variable there gives "documentation through code" that the _intention_ is to be returning the last index. The change that _would_ make sense here would be to combine the variable declaration and assignment together:
> 
> ```
> int lastIndex = static_cast<unsigned>(m_historyItems.size() - 1);
> return lastIndex;
> ```

Thanks, you are right. Then, I have one more suggestion before editing commit. If the return type of the function is **unsigned int**, shouldn't we declare the local variable as **unsigned int** also ?

```
unsigned int CalculatorHistory::AddItem(_In_ shared_ptr<HISTORYITEM> const &spHistoryItem)
{
    int lastIndex;
```

Reply:
That indeed is a great catch.  `lastIndex` _should_ be an `unsigned int`.  Given that `AddItem` is always ensuring that there is something in `m_historyItems`, the value of `m_historyItems.size()` should always be `1` or greater, meaning that we're guaranteed that our index calculation will never go negative.  Given that, yes, it should be an `unsigned int`.

Using the new tip from @janisozaur:

```suggestion
unsigned int lastIndex = static_cast<unsigned>(m_historyItems.size() - 1);
return lastIndex;
```